analyse.py

```diff
@@ -315,15 +497,20 @@ def main():
     good_messages = list(filter(good_message, all_messages))
 
     OUTDIR = "out"
+
+    # Check if the OUTDIR already exists and is not a directory
     if os.path.exists(OUTDIR):
         if not os.path.isdir(OUTDIR):
             print('Output directory "out" exists and is not a directory.')
             exit(1)
     else:
         os.makedirs(OUTDIR)
-        with open(os.path.join(OUTDIR, ".gitignore")) as f:
-            f.write("*")
 
+    # Create .gitignore file and add OUTDIR to it
+    gitignore_path = os.path.join(os.getcwd(), ".gitignore")
+    with open(gitignore_path, "a") as gitignore_file:
+        gitignore_file.write(f"{OUTDIR}/\n")

```

Comment:
Won't this add a new line to the gitignore every time you run the script?
What does this accomplish that the old approach didn't?